### PR TITLE
strings / it-academy.by / JC1-117-22 / Aliaksei Kharko / Done tasks 01-03, StringUtil and StringUtilTest

### PR DESCRIPTION
**Pull request by KharkoAO**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    public int wordCount(String str) {
+        int count = 0;
+        String[] words = str.trim().split("\\s+");
+        for (int i = 0; i < words.length; i++) {
+            if (words[i] == " ") {
+                continue;
+            } else {
+                count++;
+            }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java:110]:
```suggestion
            if (words[i] != " ") {
                count++;
            }
```

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix and add solution for task 11

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    public boolean checkPalindrome(String str) {
+        String revers = "";
+        String preparedLine = str.toLowerCase().replaceAll("\\s", "");
+        for (int i = 0; i < preparedLine.length(); i++) {
+            revers = preparedLine.charAt(i) + revers;
+        }
+        return preparedLine.equals(revers);
+    }
+
+    public String splitLine(String str, int n) {
```
[the remark this answers is not in the document]
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    public String replaceSmile(String str) {
+        return str.replace(":(", ":)");
+    }
+
+    public boolean searchWordInText(String text, String word) {
+        if (text.toLowerCase().indexOf(word.toLowerCase()) == 0 && text.toLowerCase().lastIndexOf(word.toLowerCase()) == text.length() - word.length()) {
+            return true;
+        } else {
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java (hunk 1 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    public boolean checkPalindrome(String str) {
+        String revers = "";
+        String preparedLine = str.toLowerCase().replaceAll("\\s", "");
+        for (int i = 0; i < preparedLine.length(); i++) {
+            revers = preparedLine.charAt(i) + revers;
+        }
+        return preparedLine.equals(revers);
+    }
+
+    public String splitStringApartOnNumberCharacters(String str, int n) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java:74]: method MUST return ARRAY

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

**Comment by rakovets** [comment]
outdated

**Pull request closed without merging**